### PR TITLE
Improvement on CIFAR-10 hlb_cifar10_torch.py 89%

### PR DESCRIPTION
**Pull request by g1y5x3**
Since both torch and tinygrad implemented the same model and pipeline, I started with comparing to [https://github.com/tysam-code/hlb-CIFAR10/blob/main/main.py#L274](url) as suggested. 

Two major differences:
1. Whitening layer where the weight is frozen for better data distribution
2. LR scheduler (OneCycleLR) for more efficient training

By just adding the OneCycleLR, I was able to get as high as 89% without spending too much time tuning all the parameters inside OneCycleLR, I do believe once the whitening is applied, reaching 90% should be easily reachable since the space is much cleaner.

### CODE REVIEW

**Comment by kunwar31** [comment]
@g1y5x3 currently eval is only done on a single batch of eval data, do you get the same score on entire 10k samples? (see https://github.com/geohot/tinygrad/pull/1073)

**Comment by g1y5x3** [comment]
> 

Not yet but I think that's probably why I am seeing this variation of results in between different runs. And I don't want to make too much changes to the code base yet. But I am looking into the pytorch `Dataloader` class and try to make a minimal one.

**Comment by g1y5x3** [comment]
As this was used just to compare with the tinygrad implementation which I am glad the OneCycleLR trick helped with #1073. Once the that got merged, there's probably no reason to keep this pytorch implementation anyway.
